custom_components/tuya_local/devices/fairland_x20_heatpump.yaml

```diff
@@ -0,0 +1,144 @@
+name: Pool heatpump
+products:
+  - id: a4bepadftizzrd8g
+    name: Fairland X20 series
+primary_entity:
+  entity: climate
+  dps:
+    - id: 101
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: smart
+              value: heat_cool
+            - dps_val: cool
+              value: cool
+            - dps_val: warm
+              value: heat
+    - id: 102
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: "silence"
+          value: sleep
+        - dps_val: "smart"
+          value: comfort
+        - dps_val: "booster"
+          value: boost
+    - id: 103
+      name: current_temperature
+      type: integer
+    - id: 104
+      name: temperature_unit
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: C
+        - dps_val: true
+          value: F
+    - id: 106
+      name: mode
+      type: string
+    - id: 107
+      name: temperature
+      type: integer
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: true
+              range:
+                min: 64
+                max: 104
+      range:
+        min: 18
+        max: 40
+    - id: 108
+      type: integer
+      name: min_temperature
+    - id: 109
+      type: integer
+      name: max_temperature
+    - id: 110
+      type: bitfield
+      name: error
+      mapping:
+        - dps_val: 0
+          value: OK
+        - dps_val: 4
+          value: Water Flow Protection
+          icon: "mdi:water-pump-off"
+          icon_priority: 2
+    - id: 111
+      type: bitfield
+      name: error_2
+secondary_entities:
+  - entity: sensor
+    category: diagnostic
+    class: power_factor
+    dps:
+      - id: 105
+        name: sensor
+        type: integer
+        unit: "%"
+        class: measurement
+  - entity: sensor
+    class: power
+    category: diagnostic
+    dps:
+      - id: 112
+        name: sensor
+        type: integer
+        unit: W
+        optional: true
+  - entity: binary_sensor
+    name: Water flow
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 110
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 4
+            value: true
+          - value: false
+  - entity: binary_sensor
```

Comment:
Why would this be a binary_sensor, not a switch? Same for the other two "Show..." items.

Reply:
OK, I will add the additional dps as optional attributes on the relevant entities in case they are useful to someone.